crux_time/src/lib.rs

```diff
@@ -5,24 +5,74 @@
 //! interface to do so.
 //!
 //! This is still work in progress and as such very basic.
+
 use chrono::{DateTime, Utc};
-use crux_core::capability::{CapabilityContext, Operation};
 use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+use crux_core::capability::{CapabilityContext, Operation};
+
+/// Error type for time operations
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Error)]
+pub enum TimeError {
+    #[error("invalid time")]
+    InvalidTime,
+}
+
+/// Represents a point in time:
+///
+/// - seconds: number of seconds since the Unix epoch (1970-01-01T00:00:00Z)
+/// - sub_nanos: number of nanoseconds since the last second
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Instant {
```

Comment:
I love the change to using `Instant` vs `chrono::DateTime`, since I'm using `time` in my app. It would be cool if `chrono` and `time` conversions were behind features, so you could opt-in to the ones you need and not pull in extra dependencies.

Reply:
yeah, good idea. Will do that